warn/warn_macro_test.go

```diff
@@ -0,0 +1,463 @@
+package warn
+
+import "testing"
+
+func TestUnnamedMacroNoReaderSameFile(t *testing.T) {
+	checkFindings(t, "unnamed-macro", `
+load(":foo.bzl", "foo")
+
+my_rule = rule()
+
+def macro(x):
+  foo()
+  my_rule(name = x)
+
+def not_macro(x):
+  foo()
+  native.glob()
+  native.existing_rule()
+  native.existing_rules()
+  return my_rule
+
+def another_macro(x):
+  foo()
+  [native.cc_library() for i in x]
+`,
+		[]string{
+			`5: Macro function "macro" doesn't accept a keyword argument "name".
+
+Example stack trace (statically analyzed):
+  File "test/package/test_file.bzl", line 7, in macro
+    my_rule(...)
+  File "test/package/test_file.bzl", line 3
+    my_rule = rule(...)`,
```

Comment:
I'm not convinced showing the stacktrace is really useful.

Maybe show just a line:
> "macro" is considered a macro because it calls the (rule|macro) my_rule on line 7.

Reply:
Done.